README.md

```diff
@@ -1,30 +1,37 @@
 # Jellyfish Queue
 
-The Jellyfish system processes incoming action requests and adds them to a
-queue. The system can dequeue the next action request, execute it, and post the
-results back. This module provides a small set of functions to perform any
-action request queue-related operations.
+The Jellyfish system processes incoming action requests and adds them to a queue so that they can be retrieved and executed in order.
 
-No module that interacts with the action request queue should try to bypass
-this module.
+This module provides a small set of functions to perform any queue-related operations, and should be utilized by any module requiring interaction with queued objects.
 
+Note that, although `jellyfish-queue` is mostly used for enqueuing action requests, it should be generic enough to support handling other types of objects as well. 
```

Comment:
The "should" is too strong as we haven't fully settled on that yet.

Reply:
Replaced with "could".